buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -53,6 +55,11 @@
     private static final int CLUSTER_UP_TIMEOUT = 40;
     private static final TimeUnit CLUSTER_UP_TIMEOUT_UNIT = TimeUnit.SECONDS;
 
+    private static final Semaphore globalSemaphore = new Semaphore(
+        Optional.ofNullable(System.getProperty("testclusters.max-nodes")).map(Integer::valueOf)
```

Comment:
The issue with this being `static` is that it won't be re-initialized if we change the value of the `testclusters.max-nodes` system property. We'll need a more robust way to do this if we plan to use that system property reliably. Otherwise it'll only honor that value if it's set when the daemon is first started.

Reply:
The other option is to make this "global" as a root project extension instead. We could wrap the `Semaphore` in a class, say `TestClusterLeaseRegistry`, and add that as a root project extension. Then grab the extension via `getByType()` when we need to acquire/release leases.